org.eclipse.scanning.api/src/org/eclipse/scanning/api/scan/models/ScanModel.java

```diff
@@ -216,22 +224,42 @@ public void setDetectors(IRunnableDevice<?>... detectors) {
 		this.detectors = Arrays.asList(detectors);
 	}
 	
-	public List<IScannable<?>> getMonitors() {
-		if (monitors == null) {
+	public List<IScannable<?>> getMonitorsPerPoint() {
+		if (monitorsPerPoint == null) {
+			return Collections.emptyList();
+		}
+		return monitorsPerPoint;
+	}
+	
+	public void setMonitorsPerPoint(List<IScannable<?>> monitors) {
+		logger.info("setMonitorsPerPoint({}) was {} ({})", monitors, this.monitorsPerPoint, this);
+		this.monitorsPerPoint = monitors;
+	}
+	
+	public void setMonitorsPerPoint(IScannable<?>... monitors) {
+		logger.info("setMonitorsPerPoint({}) was {} ({})", this, monitors, this.monitorsPerPoint, this);
+		this.monitorsPerPoint = new ArrayList<>(Arrays.asList(monitors));
+		for (Iterator<IScannable<?>> iterator = this.monitorsPerPoint.iterator(); iterator.hasNext();) {
+			if (iterator.next()==null) iterator.remove();
+		}
+	}
+	
+	public List<IScannable<?>> getMonitorsPerScan() {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Remove usage of generic wildcard type. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1452)